Src/AutoFakeItEasy/FakeItEasyMethodQuery.cs

```diff
@@ -74,8 +74,46 @@ public IEnumerable<ParameterInfo> Parameters
 
             public object Invoke(IEnumerable<object> parameters)
             {
-                return new Fake<T>(
-                    b => b.WithArgumentsForConstructor(parameters));
+                var genericFakeType = typeof (Fake<>).MakeGenericType(typeof (T));

```

Comment:
Would it be possible to cover these changes (lines 77 to 116) with a regression test?